src/Microsoft.ML.Data/Transforms/TypeConverting.cs

```diff
@@ -173,45 +173,45 @@ private static VersionInfo GetVersionInfo()
             /// </summary>
-            /// <param name="input">Name of input column.</param>
-            /// <param name="output">Name of output column.</param>
+            /// <param name="name">Name of the column resulting from the transformation of <paramref name="sourceColumnName"/>.</param>
             /// <param name="outputKind">The expected kind of the converted column.</param>
+            /// <param name="sourceColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="name"/> will be used as source.</param>
             /// <param name="outputKeyRange">New key range, if we work with key type.</param>
-            public ColumnInfo(string input, string output, DataKind outputKind, KeyRange outputKeyRange = null)
+            public ColumnInfo(string name, DataKind outputKind, string sourceColumnName = null, KeyRange outputKeyRange = null)
             {
-                Input = input;
-                Output = output;
+                Name = name;
+                SourceColumnName = sourceColumnName ?? name;
                 OutputKind = outputKind;
                 OutputKeyRange = outputKeyRange;
             }
         }
 
         private readonly ColumnInfo[] _columns;
 
-        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
+        private static (string outputColumnName, string sourceColumnName)[] GetColumnPairs(ColumnInfo[] columns)
         {
             Contracts.CheckNonEmpty(columns, nameof(columns));
-            return columns.Select(x => (x.Input, x.Output)).ToArray();
+            return columns.Select(x => (x.Name, x.SourceColumnName)).ToArray();
         }
 
         /// <summary>
         /// Convinence constructor for simple one column case.
         /// </summary>
         /// <param name="env">Host Environment.</param>
-        /// <param name="inputColumn">Name of the output column.</param>
-        /// <param name="outputColumn">Name of the column to be transformed. If this is null '<paramref name="inputColumn"/>' will be used.</param>
+        /// <param name="outputColumnName">Name of the output column.</param>
+        /// <param name="sourceColumnName">Name of the column to be transformed. If this is null '<paramref name="outputColumnName"/>' will be used.</param>
         /// <param name="outputKind">The expected type of the converted column.</param>
         /// <param name="outputKeyRange">New key range if we work with key type.</param>
-        public TypeConvertingTransformer(IHostEnvironment env, string inputColumn, string outputColumn, DataKind outputKind, KeyRange outputKeyRange = null)
-            : this(env, new ColumnInfo(inputColumn, outputColumn, outputKind, outputKeyRange))
+        public TypeConvertingTransformer(IHostEnvironment env, string outputColumnName, DataKind outputKind, string sourceColumnName = null, KeyRange outputKeyRange = null)
```

Comment:
>DataKind outputKind, [](start = 88, length = 20)

`outputKind` should be after `sourceColumnName`.

Reply:
no, source columnName, when nullable will come after all required params. 

---
In reply to: [251643599](https://github.com/dotnet/machinelearning/pull/2239#discussion_r251643599) [](ancestors = 251643599)

Reply:
If you look at the estimator below, the `outputKind` has default value.

---
In reply to: [251643791](https://github.com/dotnet/machinelearning/pull/2239#discussion_r251643791) [](ancestors = 251643791,251643599)

Reply:
@sfilipi 's approach seems correct since it seems like conversion argument should be required. There is a potential counterargument that when converting *to* anything, 99% of the time you're practically always converting to a float. 😄 But when it comes to something like types, I might prefer being explicit.